mitodl/micromasters
# Fix a deletion bug with the profile work history

#### What are the relevant tickets?

closes #2150 

#### What's this PR do?

For some reason we weren't passing an `index` parameter to the function to delete a work history entry, which meant that, when we called the function, the value was undefined. With that value we do something more-or-less like:

```js
const deleteEntry = index => {
  profile.work_history.splice(index, 1);
};
```

turns out that if you pass `undefined` as the first argument to `Array.prototype.splice` JavaScript will helpfully assume that you mean zero. This meant that it was possible to click the second work history entry and end up deleting the first one.

Anyway, this fixes the problem by 1) passing in the right paramter and 2) getting rid of some indirection which was originally implemented because of the user page / profile tab split but which has since become unnecessary.

#### How should this be manually tested?

Add a work history entry, and make sure that clicking the second entry in the list does not delete the first entry.

I guess also try to make sure you can repro the bug on master, to ensure it's really being fixed.
